de/factorization.py

```diff
@@ -60,7 +61,70 @@ def factorizeEstimate(data, tol=1e-9, maxiter=10000):
 
 
 def cellLineFactorization(cellLine):
+    """ Import a cell line, fit the model, and return the result. """
     data, annotation = importLINCS(cellLine)
     w, eta = factorizeEstimate(data)
+    return w, eta, annotation[0].tolist()
+
+def cellLineComparision(cellLine1, cellLine2):
+    w1, eta1, annotation1 = cellLineFactorization(cellLine1)
+    w2, eta2, annotation2 = cellLineFactorization(cellLine2)
+
+    line1_as_set = set(annotation1)
+    intersection = line1_as_set.intersection(annotation2)
+    intersection_annotation = list(intersection)
+
+    index_list1 = []
+    index_list2 = []
+
+    for x in intersection_annotation:
+        index_value1 = annotation1.index(x)
+        index_list1.append(index_value1)
+
+    for x in intersection_annotation:
+        index_value2 = annotation2.index(x)
+        index_list2.append(index_value2)
+    
+    index_list1.sort()
+    index_list2.sort()
+    return index_list1, index_list2
+
+def MatrixSubtraction(cellLine1, cellLine2):
+    w1, _, _ = cellLineFactorization(cellLine1)
+    w2, _, _= cellLineFactorization(cellLine2)
+    index_list1, index_list2 = cellLineComparision(cellLine1, cellLine2)
+
+    w1_edited = (pd.DataFrame(w1).iloc[index_list1]).values
+    w2_edited = (pd.DataFrame(w2).iloc[index_list2]).values
+
+    num_rows_W1, num_cols_W1 = w1_edited.shape
+    num_rows_W2, num_cols_W2 = w2_edited.shape
```

Comment:
Looking at the output of `w1_edited` and `w2_edited` it seems this way you are only cropping one dimension of the W matrix. 
![image](https://user-images.githubusercontent.com/34812750/123334240-a7abd900-d4f7-11eb-9864-7917e09dd832.png)

Reply:
Yep—doing this in a DataFrame is going to be tricky. If you convert to a NumPy matrix, you can just use the delete command on both axes.